share/shwap/p2p/bitswap/getter.go

```diff
@@ -74,27 +75,24 @@ func (g *Getter) Stop() {
 	g.cancel()
 }
 
-// GetShares uses [SampleBlock] and [Fetch] to get and verify samples for given coordinates.
-// TODO(@Wondertan): Rework API to get coordinates as a single param to make it ergonomic.
-func (g *Getter) GetShares(
+// GetSamples uses [SampleBlock] and [Fetch] to get and verify samples for given coordinates.
+func (g *Getter) GetSamples(
 	ctx context.Context,
 	hdr *header.ExtendedHeader,
-	rowIdxs, colIdxs []int,
-) ([]libshare.Share, error) {
-	if len(rowIdxs) != len(colIdxs) {
-		return nil, fmt.Errorf("row indecies and col indices must be same length")
+	indices []shwap.SampleCoords,
+) ([]shwap.Sample, error) {
+	if len(indices) == 0 {
+		return nil, fmt.Errorf("no sample indicies to fetch")
```

Comment:
This error seems to be common to all getters, so might worth to make it part of Getter interface

Reply:
> it part of Getter interface

you mean extract to a variable, right?

Smth like: `var ErrNoSampleIndicies = errors.New("no sample indicies to fetch")`

Reply:
yes. And add it to description of `GetSamples` method of `Getter` interface

Reply:
Done